geps/gep-1619/index.md

```diff
@@ -363,71 +433,116 @@ type SessionPersistencePolicySpec struct {
+type CookieConfig struct {
+    // LifetimeType specifies whether the cookie has a permanent or
+    // session-based lifetime. A permanent cookie persists until its
+    // specified expiry time, defined by the Expires or Max-Age cookie
+    // attributes, while a session cookie is deleted when the current
+    // session ends.
+    //
+    // When set to "Permanent", AbsoluteTimeoutSeconds indicates the
+    // cookie's lifetime via the Expires or Max-Age cookie attributes
+    // and is required.
     //
-    // Possible reasons for this condition to be False are:
+    // When set to "Session", AbsoluteTimeoutSeconds indicates the
+    // absolute lifetime of the cookie tracked by the gateway and
+    // is optional.
     //
-    // * “Invalid”
-    // * “Pending”
-    SessionPersistencePolicyConditionAccepted SessionPersistencePolicyConditionType = “Accepted”
+    // Support: Core for "Session" type
+    //
+    // Support: Extended for "Permanent" type
+    //
+    // +optional
+    // +kubebuilder:default=Session
+    LifetimeType *CookieLifetimeType `json:"lifetimeType,omitempty"`
+}
 
-    // This reason is used with the “Accepted” condition when the condition is true.
-    SessionPersistencePolicyReasonAccepted SessionPersistencePolicyConditionReason = “Valid”
+// +kubebuilder:validation:Enum=Permanent;Session
+type CookieLifetimeType string
 
-	// This reason is used with the “Accepted” condition when the SessionPersistencePolicy is invalid, e.g. crossing namespace boundaries.
-    SessionPersistencePolicyReasonInvalid SessionPersistencePolicyConditionReason = “Invalid”
+const (
+    // SessionCookieLifetimeType specifies the type for a session
+    // cookie.
+    //
+    // Support: Core
+    SessionCookieLifetimeType   CookieLifetimeType = "Session"
 
-    // This reason is used with the “Accepted” condition when the SessionPersistencePolicy is pending validation.
-    SessionPersistencePolicyReasonPending SessionPersistencePolicyConditionReason = “Pending”
+    // PermanentCookieLifetimeType specifies the type for a permanent
+    // cookie.
+    //
+    // Support: Extended
+    PermanentCookieLifetimeType  CookieLifetimeType = "Permanent"
 )
 ```
 
+### Route Rule API
+
+To support route-level configuration, this GEP also introduces an API as inline fields within the xRoute's rule section.
+The API for enabling session persistence is intentionally different from the `BackendLoadBalancePolicy` API because
+routes will have different session persistence requirements. The route rule API is expected to evolve over time. At the
+time of writing this, only `HTTPRoute` is GA, but it should also be applied to `GRPCRoute` when appropriate.
+
+```go
+type HTTPRouteRule struct {
+    [...]
+
+    // SessionPersistence defines and configures session persistence
+    // for the route rule.
+    //
+    // Support: Extended
+    //
+    // +optional
+    SessionPersistence *RouteRuleSessionPersistence `json:"sessionPersistence"`
```

Comment:
How about merging `RouteRuleSessionPersistence` and `BackendSessionPersistence`?  I think we probably need the same fields for both backend and route rule.